### PR TITLE
feat: Restrict flow log policy to use log group ARNs

### PR DESCRIPTION
**Pull request by evgenii-bezmaternykh**
## Description
This pull request updates the AWS VPC terraform module to use log group ARNs instead of wildcard ("*") as the resource in the flow log policy. This change restricts the flow log policy to specific log groups, improving security and ensuring that only authorized log groups can receive flow log data.

## Motivation and Context
The previous implementation allowed logging to any log group in AWS CloudWatch Logs, which could potentially lead to unauthorized access to log data. The updated implementation ensures that flow log data is only sent to designated log groups specified by their ARNs.

## Breaking Changes
This change does not break backwards compatibility with the current major version.

## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [x] I have executed `pre-commit run -a` on my pull request to ensure that all relevant pre-commit checks have passed.

Thank you.

### CODE REVIEW

File: vpc-flow-logs.tf [hunk 1]
```diff
@@ -8,6 +12,10 @@ locals {
   flow_log_destination_arn                  = local.create_flow_log_cloudwatch_log_group ? try(aws_cloudwatch_log_group.flow_log[0].arn, null) : var.flow_log_destination_arn
   flow_log_iam_role_arn                     = var.flow_log_destination_type != "s3" && local.create_flow_log_cloudwatch_iam_role ? try(aws_iam_role.vpc_flow_log_cloudwatch[0].arn, null) : var.flow_log_cloudwatch_iam_role_arn
   flow_log_cloudwatch_log_group_name_suffix = var.flow_log_cloudwatch_log_group_name_suffix == "" ? local.vpc_id : var.flow_log_cloudwatch_log_group_name_suffix
+  flow_log_group_arns = [
+    for log_group in aws_cloudwatch_log_group.flow_log :
+    format("arn:aws:logs:${data.aws_region.current.name}:%s:log-group:%s:*", data.aws_caller_identity.current.account_id, log_group.name)
```
Comment (bryantbiggs) [inline vpc-flow-logs.tf:17]: lets not mix format and string embedding - lets just embed the values all into the string and drop format

File: vpc-flow-logs.tf [hunk 2]
```diff
@@ -1,3 +1,7 @@
+data "aws_region" "current" {}
```
Comment (bryantbiggs) [inline vpc-flow-logs.tf:1]: put these behind the create variable - if we aren't creating anything, these APIs shouldn't be called

File: vpc-flow-logs.tf (hunk 1 above)
Comment (evgenii-bezmaternykh) [reply vpc-flow-logs.tf]: Fixed

File: vpc-flow-logs.tf (hunk 2 above)
Comment (evgenii-bezmaternykh) [reply vpc-flow-logs.tf]: Fixed, thank you.

**Review by bryantbiggs — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 5.12.0 :tada:

**Pull request merged**